vpc-sample/tests.cloudbuild.yaml

```diff
@@ -0,0 +1,108 @@
+steps:
+- id: Deploy Restricted Function
+  name: 'gcr.io/cloud-builders/gcloud'
+  dir: '${_SAMPLE_DIR}'
+  args:
+  - functions
+  - deploy
+  - restricted-${SHORT_SHA}
+  - --runtime=python38
+  - --trigger-http
+  - --no-allow-unauthenticated
+  - --ingress-settings=internal-only
+  - --entry-point=hello_world
+
+- id: 'Build Allow Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  dir: '${_SAMPLE_DIR}'
+  args: ['build', '-t', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${SHORT_SHA}', '.'] # Tag docker image with git commit sha
+
+- id: 'Push Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  args: ['push', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${SHORT_SHA}']
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: /bin/bash
+  args:
+  - '-c'
+  - | # Cloud Run Service name must be less than 63 characters
+    cat /dev/urandom | LC_CTYPE=C tr -dc 'a-z0-9' | head -c 15 > _short_id # Generate a random 15 character alphanumeric string (lowercase only)
+    gcloud alpha run deploy ${_SERVICE}-$(cat _short_id) \
+      --image gcr.io/${PROJECT_ID}/${_SERVICE}:${SHORT_SHA} \
+      --no-allow-unauthenticated \
+      --platform managed \
+      --set-env-vars=URL=https://${_SERVICE_REGION}-$PROJECT_ID.cloudfunctions.net/restricted-${SHORT_SHA} \
+      --vpc-egress=all \
+      --vpc-connector=samples-connector \
```

Comment:
Blocker: If we're pre-creating this, it should be included in the project creation script. While I wish we were testing that script, it's at least a clear source of documentation on what we think it takes to replicate the test environment.

Reply:
Added to project setup.